File: solutions/python/rotational-cipher/1/rotational_cipher.py
import string

def rotate(text, key):
    result = ""
    for char in text:
        if char.isalpha():
            char_index = string.ascii_lowercase.index(char.lower()) + key
            if char_index > 25:
                char_index -= 26
            result += string.ascii_lowercase[char_index] if char.islower() else string.ascii_lowercase[char_index].upper()
        else:
            result += char
    return result

File: solutions/csharp/rotational-cipher/1/RotationalCipher.cs
using System.Security.Principal;
using System.Threading.Channels;

public static class RotationalCipher {
  private static readonly string _letters = "abcdefghijklmnopqrstuvwxyz";
  public static string Rotate(string text, int shiftKey) {
    return string.Join("", text.Select(c => char.IsLetter(c) ? ShiftChar(c, shiftKey) : c));
  }

  private static char ShiftChar(char c, int shiftKey) {
    var charIndex = _letters.IndexOf(char.ToLower(c));
    var newCharIndex = charIndex + shiftKey;
    if (newCharIndex >= _letters.Length) {
      newCharIndex -= _letters.Length;
    }

    return char.IsUpper(c) ? char.ToUpper(_letters[newCharIndex]) : _letters[newCharIndex];
  }
}